### PR TITLE
Remove cases of six.iter* wrapped in a list()

### PR DESCRIPTION
**Pull request by funkybob**
There's absolutely no advantage [and a mild performance hit] to using six.iter*
in these cases.  The list immediately consumes the iterator.

### CODE REVIEW

File: django/contrib/admin/sites.py [hunk 1]
```diff
@@ -470,7 +470,7 @@ def index(self, request, extra_context=None):
         app_dict = self._build_app_dict(request)
 
         # Sort the apps alphabetically.
-        app_list = list(six.itervalues(app_dict))
+        app_list = list(app_dict.values())
```
Comment (MarkusH) [inline django/contrib/admin/sites.py]: I think you can drop the `list()` here.

File: django/contrib/admindocs/views.py [hunk 2]
```diff
@@ -64,7 +63,7 @@ def get_context_data(self, **kwargs):
         load_all_installed_template_libraries()
 
         tags = []
-        app_libs = list(six.iteritems(libraries))
+        app_libs = list(libraries.items())
```
Comment (MarkusH) [inline django/contrib/admindocs/views.py]: Nuke the `list()` call

File: django/contrib/admindocs/views.py [hunk 3]
```diff
@@ -97,7 +96,7 @@ def get_context_data(self, **kwargs):
         load_all_installed_template_libraries()
 
         filters = []
-        app_libs = list(six.iteritems(libraries))
+        app_libs = list(libraries.items())
```
Comment (MarkusH) [inline django/contrib/admindocs/views.py]: Here too

File: django/contrib/admindocs/views.py (hunk 2 above)
Comment (funkybob) [inline django/contrib/admindocs/views.py]: Next op with app_libs is to add it to a list...  and that is not valid.

File: django/contrib/admin/sites.py (hunk 1 above)
Comment (funkybob) [inline django/contrib/admin/sites.py]: Only because the very next line is .sort(), so I can replace the two with sorted()

**Pull request merged**